report/renderer/dom.js

```diff
@@ -154,6 +154,9 @@ export class DOM {
    * @param {string} url
```

Comment:
```suggestion
   * @param {string|undefined} url
```

Reply:
I'd prefer not littering our types with this, just because a use case comes to us from proto. If we were to go down that route, we should turn all `string` into `string|undefined` in the report.

Reply:
> we should turn all string into string|undefined in the report.

I agree :)

If you want to postpone this for a more thorough examination of *all* usage that wfm though.